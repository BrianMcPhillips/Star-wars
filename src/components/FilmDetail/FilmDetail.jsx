import React from 'react';
import PropTypes from 'prop-types';

const FilmDetail = ({ 
  title, 
  release_date,
  producer, 
  director, 
  characters,
  planets,
  species,
  starships,
  vehicles
}) => (
  <div>
    <h1>{title}</h1>
    <h3>Release date: {release_date}</h3>
    <h3>Producer: {producer}</h3>
    <h3>Director: {director}</h3>
    <h3>Characters: {characters.map((character, i) => {
      <p key={character + i}>{character}</p>;
    })}
    </h3>
    <h3>Planets: {planets.map((planet, i) => {
      <p key={planet + i}>{planet}</p>;
    })}
    </h3>
    <h3>Species: {species.map((specie, i) => {
      <p key={specie + i}>{specie}</p>;
    })}
    </h3>
    <h3>Starships: {starships.map((ship, i) => {
      <p key={ship + i}>{ship}</p>;
    })}
    </h3>
    <h3>Vehicles: {vehicles.map((vehicle, i) => {
      <p key={vehicle + i}>{vehicle}</p>;
    })}
    </h3>
  </div>
);

FilmDetail.propTypes = {
  title: PropTypes.string.isRequired,
  release_date: PropTypes.string.isRequired,
  producer: PropTypes.string.isRequired,
  director: PropTypes.string.isRequired,
  characters: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
  planets: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
  species: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
  starships: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
  vehicles: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired
};

export default FilmDetail;
