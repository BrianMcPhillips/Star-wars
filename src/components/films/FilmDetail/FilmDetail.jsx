import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const FilmDetail = ({ 
  title, 
  release_date,
  producer, 
  director, 
  characters,
  planets,
  species,
  starships,
  vehicles
}) => {
  const mapData = array => array.map((item, i) => 
    <p key={item + i}>{item}</p>
  );

  return (
    <div>
      <h1>{title}</h1>
      <h3>Release date: {release_date}</h3>
      <h3>Producer: {producer}</h3>
      <h3>Director: {director}</h3>
      <h3>Characters: {characters.map((character, i) =>
        <Link 
          to={`/char-detail/${character.split('people/')[1]}`} 
          key={character + i}>
          <p>{character}</p>
        </Link>
      )}
      </h3>
      <h3>Planets: {planets.map((planet, i) => 
        <Link
          to={`/planet-detail/${planet.split('planets/')[1]}`} 
          key={planet + i}>
        </Link>
      )}</h3>
      <h3>Species: {mapData(species)}</h3>
      <h3>Starships: {starships.map((ship, i) =>
        <Link 
          to={`/ship-detail/${ship.split('ships/')[1]}`} 
          key={ship + i}>
          <p>{ship}</p>
        </Link>
      )}</h3>
      <h3>Vehicles: {mapData(vehicles)}</h3>
    </div>
  );
};

FilmDetail.propTypes = {
  title: PropTypes.string.isRequired,
  release_date: PropTypes.string.isRequired,
  producer: PropTypes.string.isRequired,
  director: PropTypes.string.isRequired,
  characters: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
  planets: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
  species: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
  starships: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
  vehicles: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired
};

export default FilmDetail;
