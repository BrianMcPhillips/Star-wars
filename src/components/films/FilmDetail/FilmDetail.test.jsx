import React from 'react';
import { render, cleanup } from '@testing-library/react';
import FilmDetail from './FilmDetail';
import { MemoryRouter } from 'react-router-dom';

describe('FilmDetail component', () => {
  afterEach(() => cleanup());
  it('renders FilmDetail', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <FilmDetail 
          title="A New Hope" 
          director="George Lucas" 
          producer="Gary Kurtz, Rick McCallum"
          release_date= "1977-05-25" 
          characters= {[
            'https://swapi.dev/api/people/1/', 
            'https://swapi.dev/api/people/2/', 
            'https://swapi.dev/api/people/3/', 
            'https://swapi.dev/api/people/4/', 
            'https://swapi.dev/api/people/5/'
          ]}
          planets= {[
            'https://swapi.dev/api/planets/1/', 
            'https://swapi.dev/api/planets/2/', 
            'https://swapi.dev/api/planets/3/'
          ]} 
          starships= {[
            'https://swapi.dev/api/starships/2/', 
            'https://swapi.dev/api/starships/3/', 
            'https://swapi.dev/api/starships/5/', 
          ]} 
          vehicles= {[
            'https://swapi.dev/api/vehicles/4/', 
            'https://swapi.dev/api/vehicles/6/', 
            'https://swapi.dev/api/vehicles/7/', 
            'https://swapi.dev/api/vehicles/8/'
          ]} 
          species= {[
            'https://swapi.dev/api/species/1/', 
            'https://swapi.dev/api/species/2/', 
            'https://swapi.dev/api/species/3/', 
            'https://swapi.dev/api/species/4/', 
            'https://swapi.dev/api/species/5/'
          ]} 
          url="https://swapi.dev/api/films/1/"
        />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
