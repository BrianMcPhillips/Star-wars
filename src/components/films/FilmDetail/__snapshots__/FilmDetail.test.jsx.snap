// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmDetail component renders FilmDetail 1`] = `
<DocumentFragment>
  <div>
    <h1>
      A New Hope
    </h1>
    <h3>
      Release date: 1977-05-25
    </h3>
    <h3>
      Producer: Gary Kurtz, Rick McCallum
    </h3>
    <h3>
      Director: George Lucas
    </h3>
    <h3>
      Characters: 
      <a
        href="/char-detail/1/"
      >
        <p>
          https://swapi.dev/api/people/1/
        </p>
      </a>
      <a
        href="/char-detail/2/"
      >
        <p>
          https://swapi.dev/api/people/2/
        </p>
      </a>
      <a
        href="/char-detail/3/"
      >
        <p>
          https://swapi.dev/api/people/3/
        </p>
      </a>
      <a
        href="/char-detail/4/"
      >
        <p>
          https://swapi.dev/api/people/4/
        </p>
      </a>
      <a
        href="/char-detail/5/"
      >
        <p>
          https://swapi.dev/api/people/5/
        </p>
      </a>
    </h3>
    <h3>
      Planets: 
      <a
        href="/planet-detail/1/"
      >
        <p>
          https://swapi.dev/api/planets/1/
        </p>
      </a>
      <a
        href="/planet-detail/2/"
      >
        <p>
          https://swapi.dev/api/planets/2/
        </p>
      </a>
      <a
        href="/planet-detail/3/"
      >
        <p>
          https://swapi.dev/api/planets/3/
        </p>
      </a>
    </h3>
    <h3>
      Species: 
      <a
        href="/species-detail/1/"
      >
        <p>
          https://swapi.dev/api/species/1/
        </p>
      </a>
      <a
        href="/species-detail/2/"
      >
        <p>
          https://swapi.dev/api/species/2/
        </p>
      </a>
      <a
        href="/species-detail/3/"
      >
        <p>
          https://swapi.dev/api/species/3/
        </p>
      </a>
      <a
        href="/species-detail/4/"
      >
        <p>
          https://swapi.dev/api/species/4/
        </p>
      </a>
      <a
        href="/species-detail/5/"
      >
        <p>
          https://swapi.dev/api/species/5/
        </p>
      </a>
    </h3>
    <h3>
      Starships: 
      <a
        href="/ship-detail/2/"
      >
        <p>
          https://swapi.dev/api/starships/2/
        </p>
      </a>
      <a
        href="/ship-detail/3/"
      >
        <p>
          https://swapi.dev/api/starships/3/
        </p>
      </a>
      <a
        href="/ship-detail/5/"
      >
        <p>
          https://swapi.dev/api/starships/5/
        </p>
      </a>
    </h3>
    <h3>
      Vehicles: 
      <a
        href="/vehicle-detail/4/"
      >
        <p>
          https://swapi.dev/api/vehicles/4/
        </p>
      </a>
      <a
        href="/vehicle-detail/6/"
      >
        <p>
          https://swapi.dev/api/vehicles/6/
        </p>
      </a>
      <a
        href="/vehicle-detail/7/"
      >
        <p>
          https://swapi.dev/api/vehicles/7/
        </p>
      </a>
      <a
        href="/vehicle-detail/8/"
      >
        <p>
          https://swapi.dev/api/vehicles/8/
        </p>
      </a>
    </h3>
  </div>
</DocumentFragment>
`;
